src/docs/doxygen/GeosxConfig.hpp

```diff
@@ -30,7 +30,7 @@
 #define GEOSX_USE_MPI
 
 /// Enables use of OpenMP (CMake option ENABLE_OPENMP)
-/* #undef GEOSX_USE_OPENMP */
+#define GEOSX_USE_OPENMP
```

Comment:
are you sure you want to commit this file?

Reply:
the doxygen/GeosxConfig.hpp just gets added every time someone `git add -u` after doing a build that changes it, so it winds up in a lot of commits.

Reply:
There's ever chance it should be .gitignore-d but I'm not 100% on that

Reply:
We have a [way](https://github.com/GEOS-DEV/GEOS/blob/33f3f4d1feb09a0bc81ae3a0b29bc25d68a617b1/src/cmake/GeosxConfig.cmake#L70-L79) to "fix" all cmake variables used to generate this file so it doesn't diff, but the subset of variables currently covered is rather small, and as we keep adding new config macros we usually forget to extend that set.

We've decided we can't ignore this file or remove it from source control because it needs to be current for Doxygen docs to build correctly, whatever those are worth...

Reply:
Why do we even commit this file? If it's generated, it should only live in the build dir?
And when it comes to filling in the versions (e.g. for `python` or `hypre`), why don't we build a simple bootstrap executable that could be able to extract those version programmatically? `cmake` would run it during the building process.

Reply:
We commit it because `readthedocs` doesn't have the environment necessary to generate it (i.e. a compiler toolchain, `cmake` and a `thirdPartyLibs` installation).

As for versions, those are already filled by cmake directly at config time.

Reply:
> We commit it because `readthedocs` doesn't have the environment necessary to generate it (i.e. a compiler toolchain, `cmake` and a `thirdPartyLibs` installation).

Can you point me where the information contained in this file is used in `readthedocs`? I could not find it.
E.g. https://github.com/GEOS-DEV/GEOS/blob/develop/src/docs/sphinx/buildGuide/Dependencies.rst is maintained by hand (and not up to date 🤣 )

> As for versions, those are already filled by cmake directly at config time.

By parsing source files, which is not the best for sure. But whatever...

Reply:
This file is specifically for building doxygen docs hosted on `readthedocs`: https://geosx-geosx.readthedocs-hosted.com/en/latest/doxygen_output/html/_geosx_config_8hpp.html. Not sure how many people are using those, but as long as we support it, it needs to be there, and it better be generated the same way and from the same template as the one that will be generated locally in a developer's build tree.

Parsing source files is not ideal, but to build an executable to extract versions programmatically, we still need a build environment, at least a C++ compiler and a linker... then again, the only reason those version macros exist in `GeosxConfig.hpp` is because we have not implemented version extraction via runtime methods (except for MPI). This can most certainly be improved.